oca_port/squash_bot_commits.py

```diff
@@ -0,0 +1,154 @@
+import subprocess
+from .utils.misc import Output, bcolors as bc
+import click
+import tempfile
+import os
+from .utils import git as g
+
+
+MESSAGE_TO_SQUASH = [
+    "Added translation using Weblate",
+    "Translated using Weblate",
+    "Update translation files",
+]
+AUTHOR_EMAILS_TO_SQUASH = [
+    "transbot@odoo-community.org",
+    "noreply@weblate.org",
+    "oca-git-bot@odoo-community.org",
+    "oca+oca-travis@odoo-community.org",
+    "oca-ci@odoo-community.org",
+    "shopinvader-git-bot@shopinvader.com",
+]
+
+
+class SquashCommitBot(Output):
+
+    def __init__(self, app) -> None:
+        self.app = app
+        self.skipped_commits = []
+
+    def run(self):
+        if self.app.non_interactive or self.app.dry_run:
+            return False
+        click.echo(
+            click.style(
+                "🚀 Starting reducing number of commits...",
+                bold=True,
+            ),
+        )
+        squashable_commits = self._get_squashable_commits()
+        while len(squashable_commits) > 0:
+            commit = squashable_commits.pop(0)
+            result = self.squash(commit, squashable_commits)
+            if not result:
+                confirm = "Skip this commit?"
+                if click.confirm(confirm):
+                    self.skipped_commits.append(commit)
+                    print(
+                        f"\nSkipped {bc.OKCYAN}{commit.hexsha[:8]}{bc.ENDC} {commit.summary}\n"
+                    )
+            # update to get new SHAs
+            squashable_commits = self._get_squashable_commits()
+            print("\n")
+
+    def _get_squashable_commits(self):
+        commits = [
+            g.Commit(commit, addons_path=self.app.addons_rootdir, cache=self.app.cache)
+            for commit in self.app.repo.iter_commits(
+                f"{self.app.target_version}...HEAD"
+            )
+        ]
+        squashable_commits = [
+            commit
+            for commit in commits
+            if self.is_squashable_commit(commit) and not self.is_skipped_commit(commit)
+        ]
+        return squashable_commits
+
+    def squash(self, commit, squashable_commits):
+        self._print(
+            f"Squashing {bc.OKCYAN}{commit.hexsha[:9]}{bc.ENDC} {commit.summary}"
+        )
+        available_commits = [c for c in squashable_commits if c.hexsha != commit.hexsha]
```

Comment:
actually the list of squashable commits should be different from the list of commits to squash into
e.g.
- a commit from `OCA-git-bot <oca-git-bot@odoo-community.org>` that increases module's minor version
-> squashed into
- the commit that added the merged improvement